test/core.js

```diff
@@ -1,84 +1,122 @@
 
 	jQuery.migrateDeduplicateWarnings = true;
-	expectWarning( assert, "jQuery.migrateDeduplicateWarnings === true", 1, function() {
-		jQuery( "#" );
-		jQuery( "#" );
-	} );
+	expectWarning(assert, "jQuery.migrateDeduplicateWarnings === true", 1, function () {
+		jQuery("#");
+		jQuery("#");
+	});
 
 	jQuery.migrateDeduplicateWarnings = false;
-	expectWarning( assert, "jQuery.migrateDeduplicateWarnings === false", 2, function() {
-		jQuery( "#" );
-		jQuery( "#" );
-	} );
+	expectWarning(assert, "jQuery.migrateDeduplicateWarnings === false", 2, function () {
+		jQuery("#");
+		jQuery("#");
+	});
 
 	jQuery.migrateDeduplicateWarnings = origValue;
-} );
-
-QUnit.test( "jQuery(html, props)", function( assert ) {
-	assert.expect( 3 );
-
-	var $el = jQuery( "<input/>", { name: "name", val: "value", size: 42 } );
-
-	assert.equal( $el.attr( "name" ), "name", "Name attribute" );
-	assert.equal( $el.attr( "size" ),
-		jQuery.isEmptyObject( jQuery.attrFn ) ? undefined : "42", "Size attribute" );
-	assert.equal( $el.val(), "value", "Call setter method" );
-} );
-
-QUnit.test( "jQuery( '#' )", function( assert ) {
-	assert.expect( 2 );
-
-	expectWarning( assert, "Selector, through the jQuery constructor, nothing but hash", function() {
-		var set = jQuery( "#" );
-		assert.equal( set.length, 0, "empty set" );
-	} );
-} );
-
-QUnit.test( "Attribute selectors with unquoted hashes", function( assert ) {
-	assert.expect( 31 );
+});
+
+QUnit.test("jQuery(html, props)", function (assert) {
+	assert.expect(3);
+
+	var $el = jQuery("<input/>", { name: "name", val: "value", size: 42 });
+
+	assert.equal($el.attr("name"), "name", "Name attribute");
+	assert.equal($el.attr("size"),
+		jQuery.isEmptyObject(jQuery.attrFn) ? undefined : "42", "Size attribute");
+	assert.equal($el.val(), "value", "Call setter method");
+});
+
+QUnit.test("jQuery( '#' )", function (assert) {
+	assert.expect(2);
+
+	expectWarning(assert, "Selector, through the jQuery constructor, nothing but hash", function () {
+		var set = jQuery("#");
+		assert.equal(set.length, 0, "empty set");
+	});
+});
+
+QUnit.test("Improperly closed elements", function (assert) {
+	assert.expect(4);
+
+	var oldWindowAlert = window.alert;
+	window.alert = function() {
+		assert.notOk("Called alert with "+ arguments)
+	};
+
+	expectWarning(assert, "Elements not self-closable nested wrong", 4, function () {
+		jQuery("<div><p/><span/></div>");
+		jQuery("<blockquote><p class='bad' /><span/></blockquote>");
+		jQuery("<div />").append("<div data-borked='y'><span/> <p/></div>");
+		jQuery("<script /><div class=afterscript/>")
+	});
+
+	expectNoWarning(assert, "Script doesn't run in processed strings", function () {
+		jQuery("<script>alert('YOU SHOULD NOT SEE THIS ALERT')</script>");
+		jQuery("<div>").append("<script src='YOU-SHOULD-NOT-SEE-THIS.SCRIPT'></script>");
+		jQuery("<script>alert('YOU SHOULD NOT SEE THIS CONSOLE')</script>");
+		// jQuery core code causes this to execute, and also it's async
```

Comment:
I wonder if we should use this approach for jQuery 4.0 to avoid running script in the initial processing.

Reply:
This follows the native behavior, I wouldn't want to fight it too much.